fastcrypto/src/hash.rs

```diff
@@ -192,3 +194,115 @@ impl HashFunction<32> for Blake3 {
         }
     }
 }
+
+/// A Multiset Hash is a homomorphic hash function, which hashes arbitrary multisets of objects such
+/// that the hash of the union of two multisets is easy to compute from the hashes of the two multisets.
+///
+/// The hash may be computed incrementally, adding items one at a time, and the order does not affect the
+/// result. The hash of two multisets can be compared by using the Eq trait iml'd for the given hash function,
+/// and the hash function should be collision resistant.
+///
+/// See https://link.springer.com/chapter/10.1007/978-3-540-40061-5_12 for a discussion of this type of hash
```

Comment:
Please add the paper's title as well: "Incremental Multiset Hash Functions and Their Application to Memory Integrity Checking" by D. Clarke et al.